base/file.jl

```diff
@@ -513,20 +513,37 @@ function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
     return (filename, Base.open(filename, "r+"))
 end
 
+# GUID  win32 api (consider # TODO: refactor)
+struct GUID
+    l1::Culong
+    w1::Cushort
+    w2::Cushort
+    b::NTuple{8,Cuchar}
+end
+
 function tempname(parent::AbstractString=tempdir(); cleanup::Bool=true)
     isdir(parent) || throw(ArgumentError("$(repr(parent)) is not a directory"))
-    seed::UInt32 = rand(UInt32)
-    while true
-        if (seed & typemax(UInt16)) == 0
-            seed += 1
-        end
-        filename = _win_tempname(parent, seed)
-        if !ispath(filename)
-            cleanup && temp_cleanup_later(filename)
-            return filename
-        end
-        seed += 1
-    end
+
+    RPC_S_UUID_LOCAL_ONLY = 1824
+    RPC_S_UUID_NO_ADDRESS = 1739
+    RPC_S_OUT_OF_MEMORY = 14
+
+    id = Ref{GUID}()
+    r = ccall((:UuidCreate,:Rpcrt4), stdcall, Cint, (Ref{GUID},), id)
+    (r == 0 || r == RPC_S_UUID_LOCAL_ONLY) || (r == RPC_S_UUID_NO_ADDRESS && error("Cannot get Ethernet or token-ring hardware address for this computer."))
```

Comment:
```suggestion
    if !(r == 0 || r == RPC_S_UUID_LOCAL_ONLY || r == RPC_S_UUID_NO_ADDRESS) # ignore expected errors that we don't care about (and I suspect that since Vista, they aren't possible anymore)
        windowserror("UuidCreate", r % UInt32) # Throw on unexpected errors. Note, a RPC_STATUS is just a WINERROR with a different signedness of the type.
    end
```